src/schemathesis/core/jsonschema/bundler.py

```diff
@@ -135,7 +136,8 @@ def bundle_recursive(current: JsonSchema | list[JsonSchema]) -> JsonSchema | lis
 
         assert isinstance(bundled, dict)
 
-        if "$ref" in bundled and len(defs) == 1:
+        # Inlining such a schema is only possible if recursive references were inlined
+        if inline_recursive and "$ref" in bundled and len(defs) == 1:
```

Comment:
It would be nice to still do this inlining step for cases when "inline_recursive=False`, but there were no recursive references. It will make validation of responses a bit faster. Add a test for it